contracts/src/v0.8/functions/tests/1_0_0/FunctionsRouter.t.sol

```diff
@@ -3,116 +3,1346 @@ pragma solidity ^0.8.19;
+    });
+  }
+
+  function test_UpdateConfig_RevertIfNotOwner() public {
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+
+    vm.expectRevert("Only callable by owner");
+    s_functionsRouter.updateConfig(configToSet);
+  }
+
+  event ConfigUpdated(FunctionsRouter.Config config);
+
+  function test_UpdateConfig_Success() public {
+    // topic0 (function signature, always checked), NOT topic1 (false), NOT topic2 (false), NOT topic3 (false), and data (true).
+    bool checkTopic1 = false;
+    bool checkTopic2 = false;
+    bool checkTopic3 = false;
+    bool checkData = true;
+    vm.expectEmit(checkTopic1, checkTopic2, checkTopic3, checkData);
+    emit ConfigUpdated(configToSet);
+
+    s_functionsRouter.updateConfig(configToSet);
+
+    FunctionsRouter.Config memory config = s_functionsRouter.getConfig();
+    assertEq(config.maxConsumersPerSubscription, configToSet.maxConsumersPerSubscription);
+    assertEq(config.adminFee, configToSet.adminFee);
+    assertEq(config.handleOracleFulfillmentSelector, configToSet.handleOracleFulfillmentSelector);
+    assertEq(config.maxCallbackGasLimits[0], configToSet.maxCallbackGasLimits[0]);
+    assertEq(config.maxCallbackGasLimits[1], configToSet.maxCallbackGasLimits[1]);
+    assertEq(config.maxCallbackGasLimits[2], configToSet.maxCallbackGasLimits[2]);
+    assertEq(config.maxCallbackGasLimits[3], configToSet.maxCallbackGasLimits[3]);
+    assertEq(config.gasForCallExactCheck, configToSet.gasForCallExactCheck);
+  }
 }
 
 /// @notice #isValidCallbackGasLimit
-contract FunctionsRouter_IsValidCallbackGasLimit {
+contract FunctionsRouter_IsValidCallbackGasLimit is FunctionsSubscriptionSetup {
+  function test_IsValidCallbackGasLimit_RevertInvalidConfig() public {
+    // Set an invalid maxCallbackGasLimit flag
+    bytes32 flagsToSet = 0x5a00000000000000000000000000000000000000000000000000000000000000;
+    s_functionsRouter.setFlags(s_subscriptionId, flagsToSet);
+
+    vm.expectRevert(abi.encodeWithSelector(FunctionsRouter.InvalidGasFlagValue.selector, 90));
+    s_functionsRouter.isValidCallbackGasLimit(s_subscriptionId, 0);
+  }
+
+  function test_IsValidCallbackGasLimit_RevertGasLimitTooBig() public {
+    uint8 MAX_CALLBACK_GAS_LIMIT_FLAGS_INDEX = 0;
+    bytes32 subscriptionFlags = s_functionsRouter.getFlags(s_subscriptionId);
+    uint8 callbackGasLimitsIndexSelector = uint8(subscriptionFlags[MAX_CALLBACK_GAS_LIMIT_FLAGS_INDEX]);
+
+    FunctionsRouter.Config memory config = s_functionsRouter.getConfig();
+    uint32[] memory _maxCallbackGasLimits = config.maxCallbackGasLimits;
+    uint32 maxCallbackGasLimit = _maxCallbackGasLimits[callbackGasLimitsIndexSelector];
 
+    vm.expectRevert(abi.encodeWithSelector(FunctionsRouter.GasLimitTooBig.selector, maxCallbackGasLimit));
+    s_functionsRouter.isValidCallbackGasLimit(s_subscriptionId, maxCallbackGasLimit + 1);
+  }
+
+  function test_IsValidCallbackGasLimit_Success() public view {
```

Comment:
Since it runs out of gas it returns a `FulfillResult.USER_CALLBACK_ERROR`
But the `callbackReturnData` because it doesn't even have enough gas to return that.

Makes me think maybe we should have a check in there for a minimum amount of gas so that the user can at least get an error back

Reply:
Added a test case so we can observe this: `test_Fulfill_SuccessUserCallbackRunsOutOfGas`

Reply:
What I meant was just the behavior of `s_functionsRouter.isValidCallbackGasLimit(subId, 0)`.  This could be a potential test case.